android/app/src/main/java/com/zulipmobile/notifications/FcmMessage.kt

```diff
@@ -191,19 +192,7 @@ private fun extractIdentity(data: Map<String, String>): Identity =
         serverHost = data.require("server"),
         realmId = data.require("realm_id").parseInt("realm_id"),
         realmUri = data.require("realm_uri").parseUrl("realm_uri"),
-
-        // Server versions from 1.6.0 through 2.0.0 (and possibly earlier
-        // and later) send the user's email address, as `user`.  We *could*
-        // use this as a substitute for `user_id` when that's missing...
-        // but it'd be inherently buggy, and the bug it'd introduce seems
-        // likely to affect more users than the bug it'd fix.  So just ignore.
-        // TODO(server-2.0): Delete this comment.
-        // (data["user"] ignored)
-
-        // `user_id` was added in server version 2.1.0 (released 2019-12-12;
-        // commit 447a517e6, PR #12172.)
-        // TODO(server-2.1): Require this.
-        userId = data["user_id"]?.parseInt("user_id")
+        userId = data.require("user_id").parseInt("user_id")
```

Comment:
Let's leave on this a one-line comment saying it was new in 2.1.  That'll be helpful to refer to when seeing Sentry warnings about dropping FCM messages where it's absent.

(I see I didn't do this in 9620629a7571c7d981c54384cfe9f3d72c9fadbc, but I realize now I should have.)